scheduler-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/client/controller/JobsController.java

```diff
@@ -531,6 +536,47 @@ public void onSuccess(String result) {
                                       });
     }
 
+    private Request metadataRequest;
+
+    public void fetchMetadataOfPreciousResults() {
+        if (metadataRequest != null) {
+            metadataRequest.cancel();
+            metadataRequest = null;
+        }
+        Integer jobId = model.getSelectedJob().getId();
+        SchedulerServiceAsync scheduler = Scheduler.getSchedulerService();
+        metadataRequest = scheduler.getPreciousTaskName(LoginModel.getInstance().getSessionId(),
+                                                        jobId.toString(),
+                                                        new AsyncCallback<String>() {
+
+                                                            @Override
+                                                            public void onFailure(Throwable caught) {
+                                                                LogModel.getInstance()
+                                                                        .logCriticalMessage("Error while fetching metadata of precious results:\n" +
+                                                                                            JSONUtils.getJsonErrorMessage(caught));
+                                                            }
+
+                                                            @Override
+                                                            public void onSuccess(String jsonString) {
+                                                                try {
+                                                                    JSONArray array = parseJSON(jsonString).isArray();
+                                                                    List<String> preciousTaskNames = new ArrayList<>(array.size());
```

Comment:
could be improved to be one or 2 lines stream

Reply:
Hello @mboussaa , i think it is not possible due to that `array` is not suitable for streams